plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -391,9 +1394,72 @@ func (c *CiscoTelemetryMDT) parseKeyField(tags map[string]string, field *telemet
 	}
 }
 
+func (c *CiscoTelemetryMDT) parseRib(grouper *metric.SeriesGrouper, field *telemetry.TelemetryField, prefix string, path string, tags map[string]string, timestamp time.Time) {
+	// RIB
+	measurement := path
+	for _, subfield := range field.Fields {
+		//For Every table fill the keys which are vrfName, address and masklen
+		switch subfield.Name {
+		case "vrfName", "address", "maskLen":
+			tags[subfield.Name] = decodeTag(subfield)
+		}
+		if value := decodeValue(subfield); value != nil {
+			grouper.Add(measurement, tags, timestamp, subfield.Name, value)
+		}
+		if subfield.Name == "nextHop" {
+			//For next hop table fill the keys in the tag - which is address and vrfname
+			for _, subf := range subfield.Fields {
+				for _, ff := range subf.Fields {
+					switch ff.Name {
+					case "address", "vrfName":
+						key := "nextHop/" + ff.Name
+						tags[key] = decodeTag(ff)
+					}
+					if value := decodeValue(ff); value != nil {
+						name := "nextHop/" + ff.Name
+						grouper.Add(measurement, tags, timestamp, name, value)
+					}
+				}
+			}
+		}
+	}
+}
+
+func (c *CiscoTelemetryMDT) parseClassAttributeField(grouper *metric.SeriesGrouper, field *telemetry.TelemetryField, prefix string, path string, tags map[string]string, timestamp time.Time) {
+	// DME structure: https://developer.cisco.com/site/nxapi-dme-model-reference-api/
+	var nxAttributes *telemetry.TelemetryField
+	nxAttributes = field
+	isDme := strings.Contains(path, "sys/")
+	if path == "rib" {
+		//handle native data path rib
+		c.parseRib(grouper, field, prefix, path, tags, timestamp)
+		return
+	}
+	if field == nil || !isDme {
+		return
+	}
+
+	if field.Fields[0] != nil && field.Fields[0].Fields != nil && field.Fields[0].Fields[0] != nil && field.Fields[0].Fields[0].Fields[0].Name != "attributes" {
```

Comment:
Please check beforehand that `field.Fields` has at least one entry as otherwise accessing the first element will segfault!

Reply:
Looks like the code is now checking the length of `fields.Fields` is greater than zero, marking this as resolved.